sigstore/_cli.py

```diff
@@ -704,16 +696,23 @@ def _attest(args: argparse.Namespace) -> None:
 
     try:
         with open(predicate_path, "r") as f:
-            if args.predicate_type == PREDICATE_TYPE_SLSA_v0_2:
-                predicate: Predicate = SLSAPredicateV0_2.model_validate_json(f.read())
-            elif args.predicate_type == PREDICATE_TYPE_SLSA_v1_0:
-                predicate = SLSAPredicateV1_0.model_validate_json(f.read())
+            predicate = json.load(f)
+            # We do a basic sanity check using our Pydantic models to see if the
+            # contents of the predicate file match the specified predicate type.
+            # Since most of the predicate fields are optional, this only checks that
+            # the fields that are present and correctly spelled have the expected
+            # type.
+            if args.predicate_type == PredicateType.SLSA_v0_2.value:
+                SLSAPredicateV0_2.model_validate(predicate)
+            elif args.predicate_type == PredicateType.SLSA_v1_0.value:
+                SLSAPredicateV1_0.model_validate(predicate)
             else:
                 _invalid_arguments(
                     args,
-                    f'Unsupported predicate type "{args.predicate_type}". Predicate type must be one of: {SUPPORTED_PREDICATE_TYPES}',
+                    f'Unsupported predicate type "{args.predicate_type}". Predicate type must be one of: {[p.value for p in PredicateType]}',
```

Comment:
With the suggestion below, I think this can just be:

```suggestion
                    f'Unsupported predicate type "{args.predicate_type}". Predicate type must be one of: {list(PredicateType)}',
```

(but you should sanity check that 😅)

Reply:
Fixed